### PR TITLE
Job: new SMN UI

### PR DESCRIPTION
**Pull request by Echoring**
SMNUI
### **Following function has been added:**
- **Aetherflow Box**
Show how many Aetherflow Stack you have.
Turn red when you still have 2 stack and Energy Drain is about to ready in 8s(for finish current cast, then cast a Fester).
- **Demi-Summoning Box**
When not in Demi-Summoning/Trancing, show what you can summon next turn by color.
While Summoning/Trancing, show time remains.
Turn red when Demi-Summon time remains less than 7s, to alarm player cast the second Enkindle, and don't cast a spell which has cast time. 
Also turn red when Trance time remain less than 2s for Deathflare.
- **Miasma & Bio DoT Tracker**
To track 2 DoT, Tracker part looks a little crowded.
- **Energy Drain Cooldown Tracker**
Both Energy Drain and Energy Siphon is supported.
- **Trance Cooldown Tracker**
Infact, Trance's Cooldown is 55s, but wait till 60s is better on matching Raid buffs.
A long Threshold at about 9s is set for those who want to use 55s rotation.

### CODE REVIEW

**Comment by quisquous** [comment]
Thanks for this.  It will probably take me a week or so to get back to this, as there's a lot of new patch content to look at.

**Comment by Echoring** [comment]
### **New feature:**
- **FurtherRuin Stack Guage**
r4
Knowing issue:
If you lose "Further Ruin" buff without an effectlose log like change zone or change job, it will show error stacks.
But it will be corrected at once when you gain your first "Further Ruin" next fight.
(I'm not sure is there anyway to set furtherRuin to 0 upon zone or job changed)

File: ui/jobs/jobs.js [hunk 1]
```diff
@@ -2477,6 +2655,9 @@ class Bars {
     this.UpdateFoodBuff();
     if (this.buffTracker)
       this.buffTracker.clear();
+
+    for (let i = 0; i < this.changeZoneFuncs.length; ++i)
```
[the remark this answers is not in the document]
Comment (valarnin) [reply ui/jobs/jobs.js:2659]: I was following the patterns elsewhere in the file when I suggested that format, by the way. E.g. https://github.com/quisquous/cactbot/blob/53de4407a193f7601bb7011f07d0cebf513946e3/ui/jobs/jobs.js#L2756-L2757

Wonder if there's an eslint for that?

[the remark this answers is not in the document]
Comment (quisquous) [reply ui/jobs/jobs.js:2659]: I think I've looked for one before and haven't found it.

**Comment by Echoring** [comment]
Done! Also change `toFixed(0)` in setupSch to `Math.floor()` where I once write.
Thank you both very much!

**Comment by quisquous** [comment]
Thank you!

**Pull request merged**